Added port to modbus tcp config

sploders101 opened the pull request:
self._clients has keys in the format of "{{host}}:{{port}}" now, and a port field has been added to the block configuration

sploders101 committed:
Added port to modbus tcp config

shadetree01010100 reviewed (changes requested):
Thank you for the Pull Request! This is a great feature to add, but a few changes will be needed before we can merge this.

In addition to updating spec.json, README.md needs to be updated to describe the new property, as well. You can use the CLI tool `nio buildreadme` from the block directory to do this automatically.

The unit tests also need to be updated, currently they cannot pass. If you are not familiar with unit testing in python, pytest is a commonly used runner https://docs.pytest.org/en/latest/ To use it you need to run `pytest tests` from the block directory, and pytest will discover and execute the tests insides `tests/`. if you try this with your branch, 8 of the 19 tests fail. The block test case is defined here https://github.com/niolabs/nio/blob/master/nio/testing/block_test_case.py and provides some helpful assertions in addition to those inherited from https://docs.python.org/3/library/unittest.mock.html

The test should be updated to cover the new property, both if the block is configured with a static `502` for example, and also if it evaluates with the incoming signal, for example `{{ $port }}` where the incoming signal includes the attribute `port` with an integer value, similar to what's happening inside `test_multiple_tests` here https://github.com/nio-blocks/modbus/blob/master/tests/test_modbus_tcp_block.py#L74

sploders101 commented:
Sounds good! I am in the service department today, so I'll make these changes and resubmit this evening

sploders101 committed:
Fixed test cases

sploders101 committed:
Fixed reconnect bug

sploders101 commented:
I used the nio updatereadme tool to update the readme, and fixed the bug with reconnecting. 
I had attempted to use the properties of the client itself to avoid string manipulation with the index.
```python
self._connect_to_host(self._clients[host].host, self._clients[host].port)
```
However, I must have misunderstood pymodbus3's code, because those properties don't exist. I updated the branch with a version that splits the index, and converts the port to an int before passing them on.

shadetree01010100 commented on modbus_tcp_block.py:
```diff
@@ -116,8 +116,9 @@ def _connect(self, host=None, port=502):
-            for host in self._clients:
-                self._connect_to_host(self._clients[host].host, self._clients[host].port)
+            for index in self._clients:
+                indexVars = index.split(":")
```
in Python you can unpack an iterable, such as returned by `split`, into multiple values, like this:
```
for client in self._clients:
    host, port = client.split(":")
    self._connect_to_host(host, int(port))
```

shadetree01010100 reviewed (approved):
This looks great, thanks for taking the time to contribute! I left a suggestion for your reference as you learn more Python, but the code works great!
